sql_joins_demystified/sql_joins_demystified.md

```diff
@@ -0,0 +1,589 @@
+| 41356   | 2022-02-10   | 1    | 3  | 2    | 3    | 10   |
+
+**Table 2: `subject_address`**
+
+<!-- data-type="none" class="tight-table" style="font-size:80%" -->
+| subj_id  | street_address  | city  | state   | zip  | date_start  | date_end   |
+| :--------- | :--------- | :--------- | :--------- | :--------- | :--------- | :--------- |
+| 11234   | 123 Main Street   | Smithtown    | PA  | 19000    | 2022-01-01   | `NULL`    |
+| 11234   | 123 Oak Lane   | Old Towne    | PA  | 18000   | 2000-01-01    | 2021-12-31    |
+| 93452   | 123 Green Blvd  | Kirby    | TN  | 37000    | 2020-05-01    | `NULL`   |
+
+Notice the one-to-many aspects of this data.  The subject with id 86234 has two different administrations of the depression inventory, on different dates.  The subject with id 11234 has two different addresses for two different date ranges.
+
+**Column Conundrums**
+
+Imagining how to consolidate this data into one single table with one row per subject is challenging.  Would we provide two sets of columns for addresses?  Maybe the column names would be "street\_address\_1", "city\_1", etc., and "street\_address\_2", "city\_2", and so on.  But this is a longitudinal study and people might move 3, 5, even 10 times! The same problem emerges with multiple administrations of the depression inventory.  If we try to combine data that can repeat into a "one row per subject" model, we don't know how many columns to create.
+
+**Row Woes**
+
+And it gets worse.  We don't just have to worry about columns, but also rows.  Consider the fact that we also have other, unchanging data about subjects that we'd need to include, like their demographic information (date of birth, sex, race, etc.).  If someone has four administrations of the depression inventory, do we list their sex, race, and date of birth on all four of those administrations?  That's a lot of data repetition which takes up disk space and also opens up more possibilities for errors.
+
+**One Table?  Not a Good Idea!**
+
+In short, if we try to force all the data we care about into a single table, we have a problem of rapidly multiplying columns and data repetition in rows.  This makes maintaining, cleaning, correcting, and understanding data very tricky.
+
+We're much better off collecting various groups of facts about our research subjects into distinct, separate tables, like a "demographics" table, a "depression" table, an "anxiety" table, an "address" table, and so on, including a "subj\_id" column to link rows back to the subject in question.  This realization, of how we can separate data into concepts that each get a table, leads us to the idea of **normalization**.
+
+## Normalization
+
+When data is normalized, we organize data to reduce the possibility of needless data duplication and empty cells.  Normalization is a complex information science topic and has different degrees of completeness, so here we'll talk about what normalization **looks like** more than what it **is**.  
```

Comment:
It might feel repetitive from the last line of the previous page, but because they are separate pages, and in fact separate sections, it would be helpful to repeat the definition of **normalization** here. e.g.

"Normalization is the process of separating data into concepts, each of which is given its own table." or something like that.